### DISCUSSION TITLE
How to close the lightbox from within it?

### DISCUSSION THREAD

**Discussion by ewuski** [biati-digital/glightbox#484] [Q&A]
I want to close the lightbox from within it, not by clicking the close button.

Either I wanna show a button within the lightbox body that upon clicking closes the lightbox. Or add a listener within the lightbox that closes the lightbox when activated.

Any idea how to do either of these or both? Can't find anything in the docs. Thanks

**Comment by gingerchew** [comment]
You would need to add the listener on open, the html is generated on the fly

**Reply by ewuski** [reply]
Yeah, that's the problem - which one?

I need to trigger the close event on an element within the lightbox that is created on the fly... So when the lightbox is being opened, the element does not yet exist, so I am unable to add the listener to it.

**Reply by gingerchew** [reply]
Right, the open event.

```js
const lb = GLightbox();

lb.on('open', () => yourEventHandler());
```

**Comment by ewuski** [comment]
No way I could make it work by setting up a listener on opening but I made it work another way.

However, I have an issue displaying the content of the popup if the anchor that triggers it has a title attribute:

`<a href="/my-page" title="Anchor title" class="glightboxLogin">My page</a>`

Upon clicking, the slide only displays "Anchor title" and that's it. It does not display the `/my-page` content at all.

While this:

`<a href="/my-page" class="glightboxLogin">My page</a>`

displays the page content correctly in the popup.

I am unable to find in the docs how to stop the title from replacing the page content. Any suggestion?

**Reply by gingerchew** [reply]
You probably want to use the `data-type` attribute. specifically `data-type="external"`.

**Comment by ewuski** [comment]
Like this?

`<a href="/my-page" title="Anchor title" class="glightboxLogin" data-type="external">My page</a>`

It doesn't have any effect. Still displays only "Anchor title" in the popup.
